### ISSUE TITLE
linter: auto discover system tsgolint bin (using mise)

### ISSUE THREAD

**Issue by rubnogueira** [oxc-project/oxc#14594]
### What version of Oxlint are you using?

1.23.0

### What command did you run?

oxlint --type-aware

### What does your `.oxlintrc.json` config file look like?

Nothing

### What happened?

I'm using mise (which is a tool manager to install java, node, php, etc) to install oxlint.

It works perfectly but it fails to detect tsgolint installed with mise.

How to repro:
```
mise use -g "npm:oxlint"
mise use -g "npm:@typescript/native-preview@latest"
mise use -g "npm:oxlint-tsgolint@latest"

❯ oxlint -V
Version: 1.23.0

❯ tsgolint --help
✨ tsgolint - speedy TypeScript linter

Usage:
    tsgolint [OPTIONS]

Options:
    --tsconfig PATH   Which tsconfig to use. Defaults to tsconfig.json.
                --list-files      List matched files
    -h, --help        Show help
``` 

But then when I use the --type-aware flag:

```
> oxlint --type-aware
Failed to find tsgolint executable

> which tsgolint
/Users/myuser/.local/share/mise/installs/npm-oxlint-tsgolint/0.2.0/bin/tsgolint
```

If I define the OXLINT_TSGOLINT_PATH variable, it works, but ideally oxlint should find that path, since it's available under "tsgolint".

Context:
https://discord.com/channels/1079625926024900739/1080723403595591700/1427583511484694579

**Comment by magic-akari** [comment]
Hi @rubnogueira , Thanks for reporting this! I suspect this might be related to how mise handles package installations. Similar to the Prettier plugin issue in [jdx/mise#5521](https://github.com/jdx/mise/discussions/5521), mise installs each npm package in isolated directories rather than the traditional node_modules/.bin structure. Additionally, oxlint-tsgolint is listed as a peerDependency of oxlint, and it's unclear how mise handles peer dependencies. 

Question for the Oxc team: Should we add system `PATH` lookup as a fallback? This would help with package managers like mise that use shims, but wanted to check if this fits the project's design. For now, setting `OXLINT_TSGOLINT_PATH` should work as a workaround.

**Comment by camc314** [comment]
> Should we add system PATH lookup as a fallback? 

yes, I would be a good fallback to add. Although, possibly tricky to test correctly
